core/src/main/java/org/testcontainers/lifecycle/Startables.java

```diff
@@ -0,0 +1,50 @@
+package org.testcontainers.lifecycle;
+
+import lombok.experimental.UtilityClass;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Stream;
+
+@UtilityClass
+public class Startables {
+
+    private static final Executor EXECUTOR = Executors.newCachedThreadPool(new ThreadFactory() {
+
+        private final AtomicLong COUNTER = new AtomicLong(0);
+
+        @Override
+        public Thread newThread(Runnable r) {
+            Thread thread = new Thread(r, "testcontainers-lifecycle-" + COUNTER.getAndIncrement());
+            thread.setDaemon(true);
+            return thread;
+        }
+    });
+
+    public CompletableFuture<Void> deepStart(Collection<Startable> startables) {
+        return deepStart(startables.stream());
+    }
+
+    public CompletableFuture<Void> deepStart(Stream<Startable> startables) {
+        return deepStart(new HashMap<>(), startables);
+    }
+
+    private CompletableFuture<Void> deepStart(Map<Startable, CompletableFuture<Void>> started, Stream<Startable> startables) {
```

Comment:
This is very nice!

Again I think some comments might help. e.g. there are more naive ways of doing this that would be easier code that would not be able to work asynchronously as this does, so it's worth explaining that the intent of these methods is to be as parallelized as possible.